### PR TITLE
Filter duplicates

### PR DESCRIPTION
**Pull request by Alan32Liu**
1. Update CF dependency to `2.6.0` to use `crash comparer`.
2. Replace the naive `duplicated` function from `pandas` with `crash_comparer` from `clusterfuzz`.

### CODE REVIEW

File: analysis/data_utils.py [hunk 1]
```diff
@@ -128,17 +129,37 @@ def filter_max_time(experiment_df, max_time):
     return experiment_df[experiment_df['time'] <= max_time]
 
 
+def is_unique_crash(crash_group):
+    """Check if each crash in |crash_group| is unique with CF's crash comparer.
+    Return the |crash_group| with an extra columns representing if that crash
+    is the first occurrence."""
+    unique_crashes = set()
+    is_firsts = []
+    for crash in crash_group.crash_key:
```
Comment (oliverchang) [inline analysis/data_utils.py:138]: `crash_key` is the concatenation of both the crash type and the crash state: https://github.com/google/fuzzbench/blob/e2761292572de489049aa64d518044f660578f59/experiment/measurer/run_crashes.py#L101

This is inconsistent with CF's grouping behaviour, which does this based only on the crash state (assuming the security relevance is the same). 

We should do a split on ":" here to only consider the crash state. This doesn't take into account the security relevance, but I think that's fine for the purposes of benchmarking.

**Review by oliverchang — COMMENTED** [review]
thanks! also are there any tests we can add to https://github.com/google/fuzzbench/blob/master/analysis/test_data_utils.py ?

**Comment by Alan32Liu** [comment]
Testing this with experiment [`2023-02-13-test-bug-grouping`](https://pantheon.corp.google.com/storage/browser/fuzzbench-data/2023-02-13-test-bug-grouping;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false).

**Comment by oliverchang** [comment]
Thanks! Is it possible to also add a unit test to https://github.com/google/fuzzbench/blob/master/analysis/test_data_utils.py ?

**Comment by Alan32Liu** [comment]
> Thanks! Is it possible to also add a unit test to https://github.com/google/fuzzbench/blob/master/analysis/test_data_utils.py ?

Good point, thanks! Adding one now.

**Comment by Alan32Liu** [comment]
> Thanks! Is it possible to also add a unit test to https://github.com/google/fuzzbench/blob/master/analysis/test_data_utils.py ?

Added two testcases (`bug_experiment_{1,2}_df.csv`) with data generated by a 15-minute experiment and a 5-hour one, respectively.
`expected_{1,2}.csv` are modified based on the old method, and some manual test data are also added to `expected_2.csv` to capture more cases.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Review by oliverchang — APPROVED** [review]

**Comment by Alan32Liu** [comment]
> Testing this with experiment [`2023-02-13-test-bug-grouping`](https://pantheon.corp.google.com/storage/browser/fuzzbench-data/2023-02-13-test-bug-grouping;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false).

For some reason, this experiment did not produce any report and barely left any data in the experiment data storage.

I launched another 1-hour experiment with the latest master, [the result](https://storage.googleapis.com/www.fuzzbench.com/reports/2023-02-15-test-bug-grouping-1/index.html) seems to be fine.

**Comment by jonathanmetzman** [comment]
> > Testing this with experiment [`2023-02-13-test-bug-grouping`](https://pantheon.corp.google.com/storage/browser/fuzzbench-data/2023-02-13-test-bug-grouping;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false).
> 
> For some reason, this experiment did not produce any report and barely left any data in the experiment data storage.
> 
> I launched another 1-hour experiment with the latest master, [the result](https://storage.googleapis.com/www.fuzzbench.com/reports/2023-02-15-test-bug-grouping-1/index.html) seems to be fine.

We really should be testing this with a full experiment though,  not just a toy one.

**Comment by Alan32Liu** [comment]
`2023-02-15-test-bug-grouping-2`

> > > Testing this with experiment [`2023-02-13-test-bug-grouping`](https://pantheon.corp.google.com/storage/browser/fuzzbench-data/2023-02-13-test-bug-grouping;tab=objects?project=fuzzbench&prefix=&forceOnObjectsSortingFiltering=false).
> > 
> > 
> > For some reason, this experiment did not produce any report and barely left any data in the experiment data storage.
> > I launched another 1-hour experiment with the latest master, [the result](https://storage.googleapis.com/www.fuzzbench.com/reports/2023-02-15-test-bug-grouping-1/index.html) seems to be fine.
> 
> We really should be testing this with a full experiment though, not just a toy one.

`2023-02-15-test-bug-grouping-2` is running, should be able to get the results in 24 hours.
I did not expect this to take 3 days: I should have used non-preemptible on this as well.

**Pull request merged**